### ISSUE TITLE
Can't login after password change

### ISSUE THREAD

**Issue by ygoe** [git-for-windows/git#1761]
tl;dr:

I've recently changed my password for GitLab.com and now tried to clone one of my private repositories. That didn't work because I got a failed authentication message. I can retry as I might, the message won't go away. But nobody is actually asking me for the new password and I can't find the place where it is stored.

So my best guess right now is to reinstall Windows to get rid of the saved password somewhere hidden and be able to clone my repositories again. Of course this isn't the best option as you might imagine. So I'm looking for a better solution to update my saved password.

Normally the client software would ask me for a password if it was rejected. But Git can't do that. I consider this a bug that needs to be fixed.

------

 - [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

64 bit

```
$ git --version --build-options

git version 2.17.1.windows.2
cpu: x86_64
built from commit: a60968cf435951d9411fc0f980a2e362d5cccea2
sizeof-long: 4
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.16299.547]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

defaults
```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

** insert your machine's response here **
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD, TortoiseGit

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
Clone repository with TortoiseGit via HTTP
```
 - What did you expect to occur after running these commands?

Get a cloned repository

 - What actually happened instead?

Could not authenticate at gitlab.com for my private repository because I recently changed my GitLab account password.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** insert URL here **

**Comment by lhasadad** [comment]
you might want to check the progit doc on git-scm.   there is a section on about page 333 of the pdf called: Credential Storage.  chances are you did something like what they suggest and that doc might help you find/change the pw.

**Comment by shiftkey** [comment]
@ygoe Git defers to a credential helper to manage it's credentials:

```
$ git config credential.helper
```

This corresponds to a program named `git-credential-{helper-name}` (by convention) as part of the Git installation. Can you check what you have installed currently, as this will help out with troubleshooting.

@lhasadad this is the corresponding page on the Git SCM website: https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage

The details of this page are important because Git should be invalidating the credentials (using the `erase` action) if it finds the credentials don't work for the host, so maybe the credential helper isn't following that action.

**Comment by dscho** [comment]
@ygoe for the record: everyone here waited for your answer.

**Comment by ygoe** [comment]
Sorry for wasting a few weeks of your life time spent on waiting for an answer to a question never asked.

I didn't want to write a custom credential manager which that referenced page seems to explain to me. Actually, I think I've already clarified it all. That default credential manager for Windows is installed automatically. Does it have a name? It's simply the default one. It doesn't seem to work correctly. I just wanted to report that. Maybe you're interested in knowing so you could fix it. I certainly can't. Luckily I'm not changing passwords too often and Windows 10 still has a little remnant of the old control panel that lets me delete stored credentials. But that might go away one day.

If you need to know anything else, please tell me what to do.
